<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro">

	<!-- ================================ Robot Base ================================ -->
	<!-- The xacro macro xacro: EISpider_base contains: base and trunk -->
	<xacro:macro name="EISpider_base" >
		<!-- Floating-base Joint -->
		<joint name="jnt_floating_base" type="fixed">
			<origin xyz="0 0 0" rpy="0 0 ${PI/2}"/>
			<parent link="link_base"/>
			<child link="link_trunk"/>
		</joint>
		<!-- Links -->
		<!-- Footprint link -->
		<link name="link_base">
			<visual>
				<geometry>
					<cylinder length="0.01" radius="0.01"/>
				</geometry>
                <material name="bbbb">
                    <color rgba="0.8 0.8 0.8 1"/>
                </material>
			</visual>
		</link>
		<!-- Base link -->
        <link name="link_trunk">
            <visual>
                <origin xyz="-0.4264 -0.48 -0.07" rpy="${PI/2} 0 ${PI/2}"/>
                <geometry>
                    <mesh filename="package://hit_spider/model/meshes/base_1.stl"   scale="0.001 0.001 0.001"/>
                </geometry>
                <material name="bbbb"/>
            </visual>

            <visual>
                <origin xyz=" -0.350 -0.315 0.22" rpy="${PI/2} 0 ${PI/2}"/>
                <geometry>
                    <mesh filename="package://hit_spider/model/meshes/base_2.stl"   scale="0.001 0.001 0.001"/>
                </geometry>
                <material name="bbbb"/>
            </visual>

            <collision>
                <origin xyz="0 0 0.1" rpy="0 0 0"/>
                <geometry>
                    <cylinder length="0.3" radius="0.3"/>
                </geometry>
            </collision>

            <inertial>
                <origin xyz="0 0 0.1" rpy="0 0 0"/>
                <mass value="${base_mass}"/>
                <inertia
                ixx="${base_ixx}" ixy="0.0" ixz="0.0"
                iyy="${base_iyy}" iyz="0.0"
                izz="${base_izz}"/>
            </inertial>
        </link>

     <!-- <link name="camera_link">
        <visual>
            <origin xyz=" 0 0 0 " rpy="0 0 ${PI/2}" />
            <geometry>
                <mesh filename="/home/lzy/t_p/src/hitSpider/meshs/sensors/kinect.dae"/>
            </geometry>
            <material name="black">
                <color rgba="0 0 0 0.95"/>
            </material>
        </visual>
    </link>
 
    <joint name="camera_joint" type="fixed">
        <origin xyz="0.0 0.25 0.32" rpy="0 0 ${PI/2}"/>
        <parent link="link_base"/>
        <child link="camera_link"/>
    </joint> -->

    <!-- <link name="laser_link">
        <visual>
            <origin xyz=" 0 0 0 " rpy="0 0 0" />
            <geometry>
                <mesh filename="/home/lzy/t_p/src/hitSpider/meshs/sensors/kinect.dae"/>
            </geometry>
            <material name="black">
                <color rgba="0 0 0 0.95"/>
            </material>
        </visual>
    </link>
 
    <joint name="camera_joint" type="fixed">
        <origin xyz="0.0 0.25 0.32" rpy="0 0 ${PI/2}"/>
        <parent link="link_base"/>
        <child link="camera_link"/>
    </joint> -->

        <!--/IMU sensor gazebo plugin-->
        <xacro:include filename="$(find hit_spider)/model/xacro/base/base.gazebo.xacro"/>
        <xacro:IMUsensor  load_imu_sensor="true" link_name="link_base" x="0" y="0" z="0"/>
        </xacro:macro>



</robot>
